user-interface/src/main/java/life/qbic/datamanager/views/general/spreadsheet/Spreadsheet.java

```diff
@@ -615,6 +631,28 @@ private int colIndex(Column<T, ?> column) {
     return columns.get(colIndex);
   }
 
+  public void hideColumn(Column<T, ?> column) {
+    if (!columns.contains(column)) {
+      throw new IllegalArgumentException(
+          "Cannot hide column. The column is not part of this spreadsheet");
+    }
+    column.hide();
+    delegateSpreadsheet.setColumnHidden(columns.indexOf(column), column.isHidden());
+  }
+
+  public void showColumn(Column<T, ?> column) {
+    if (!columns.contains(column)) {
+      throw new IllegalArgumentException(
+          "Cannot show column. The column is not part of this spreadsheet");
+    }
+    column.show();
+    delegateSpreadsheet.setColumnHidden(columns.indexOf(column), column.isHidden());
+  }
+
+  public void lockColumn(Column<T, ?> column) {
+    column.withCellStyle(lockedCellStyle);
+  }
```

Comment:
I'd suggest to be consistent and also throw an IllegalArgumentException here if the column is not part of the spreadsheet.  🤔